homeassistant/components/denonavr/receiver.py

```diff
@@ -60,19 +73,19 @@ async def async_connect_receiver(self):
 
         return True
 
-    def init_receiver_class(self):
-        """Initialize the DenonAVR class in a way that can called by async_add_executor_job."""
+    async def async_init_receiver_class(self) -> bool:
+        """Initialize the DenonAVR class asynchronously."""
+        receiver = DenonAVR(
+            host=self._host,
+            show_all_inputs=self._show_all_inputs,
+            timeout=self._timeout,
+            add_zones=self._zones,
+        )
+        # Use httpx.AsyncClient getter provided by Home Assistant
+        receiver.set_async_client_getter(self._async_client_getter)
         try:
-            self._receiver = denonavr.DenonAVR(
-                host=self._host,
-                show_all_inputs=self._show_all_inputs,
-                timeout=self._timeout,
-                add_zones=self._zones,
-            )
-        except ConnectionError:
-            _LOGGER.error(
-                "ConnectionError during setup of denonavr with host %s", self._host
-            )
-            return False
+            await receiver.async_setup()
+        except (AvrNetworkError, AvrTimoutError) as ex:
```

Comment:
Sorry I missed that this is being used in the config flow as well.  

It would be cleaner to let the exception raise and catch both of them in Init and throw configentrynotready there and catch them in the config flow instead of 
Configentrynotready


This looks good. Once the coverage is fixed, should be good to merge 👍